src/phases/encounter-phase.ts

```diff
@@ -220,6 +216,18 @@ export class EncounterPhase extends BattlePhase {
       if (!this.loaded && battle.battleType !== BattleType.MYSTERY_ENCOUNTER) {
         regenerateModifierPoolThresholds(this.scene.getEnemyField(), battle.battleType === BattleType.TRAINER ? ModifierPoolType.TRAINER : ModifierPoolType.WILD);
         this.scene.generateEnemyModifiers();
+        // This checks if the current battle is an Endless E-Max battle/Classic final boss and sets the MBH held by the boss to untransferrable
+        if (this.scene.currentBattle.waveIndex % 1000 || battle.battleSpec === BattleSpec.FINAL_BOSS) {
+          const enemyPokemon = this.scene.getEnemyPokemon();
+          if (enemyPokemon) {
+            const bossMBH = this.scene.findModifier(m => m instanceof TurnHeldItemTransferModifier && m.pokemonId === enemyPokemon.id, false) as TurnHeldItemTransferModifier;
+            if (bossMBH) {
+              this.scene.removeModifier(bossMBH!);
+              bossMBH?.setTransferrableFalse();
+              this.scene.addEnemyModifier(bossMBH!);
```

Comment:
These `!` and the `?` shouldn't be necessary anymore.

```suggestion
              this.scene.removeModifier(bossMBH);
              bossMBH.setTransferrableFalse();
              this.scene.addEnemyModifier(bossMBH);
```

 
_WE don't want to add new bangs. They were merely a desperate measure to speed up the adoption of strict-null_